docs/basics/101-109-rerun.rst

```diff
@@ -0,0 +1,179 @@
+           author = basename(file).split('__')[1] \
+               if len(basename(file).split('__')[1].split('_')) > 1 \
+               else '__'.join(basename(file).split('__')[1:3])
+           # extract title
+           title = '_'.join(basename(file).split(author)[-1].split('__')[1:])
+           # write speaker and talk title into the file
+           f.write(author + '\t' + title + '\n')
+   EOT
+
+Because the script is now modified, save the modifications to the dataset.
+We can use the shorthand "BF" to denote "Bug fix" in the commit message.
+
+.. runrecord:: _examples/DL-101-108-111
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ datalad status
+
+.. runrecord:: _examples/DL-101-108-112
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ datalad save -m "BF: list both directories content" code/list_titles.py
+
+What we *could* do is to run the same ``datalad run`` command as before to recreate
+the file, but now with all of the contents:
+
+.. code-block:: bash
+
+   # don't execute this!
+   $ datalad run -m "create a list of podcast titles" "python code/list_titles.py"
+
+However, think about any situation where the command would be longer than this,
+or that is many months past the first execution. It wouldn't be easy to remember
+the command, nor would it be very convenient to copy it from the ``run-record``.
+
+Luckily, a fellow student remembered the DataLad way of re-executing
+a ``run`` command, and he's eager to show it to you.
+
+"In order to re-execute a ``datalad run`` command,
+find the checksum of its commit and use it as an argument for the
+``datalad rerun`` command! That's it!" he says happily.
+
+So you go ahead and find the commit checksum in your history:
+
+.. runrecord:: _examples/DL-101-108-113
+   :language: console
+   :workdir: dl-101/DataLad-101
+   :lines: 1-12
+   :emphasize-lines: 8
+
+   $ git log -2
+
+Take that checksum and paste it after ``datalad rerun``
+(the first 6-8 characters of the checksum would be sufficient,
+here we're using all of them).
+
+.. runrecord:: _examples/DL-101-108-114
+   :language: console
+   :workdir: dl-101/DataLad-101
+   :realcommand: echo "$ datalad rerun $(git rev-parse HEAD~1)" && datalad rerun $(git rev-parse HEAD~1)
+
+Now DataLad has made use out of the ``run-record``, and based on the information in
+it re-executed the original command. Because we
+updated the script, the output ``Podcasts.tsv`` has changed and now contains the podcast
+titles of both subdirectories.
+You've probably already guessed it, but the easiest way to check whether a ``datalad rerun``
+has changed the desired output file is
+to check whether the rerun command appears in the datasets history: If a ``datalad rerun``
+does not add or change any content in the dataset, it will also not be recorded in the history.
+
+.. runrecord:: _examples/DL-101-108-115
+   :language: console
+   :workdir: dl-101/DataLad-101
+   :emphasize-lines: 4
+
+   git log -1
```

Comment:
It might be useful to introduce `datalad diff` and `git diff` here...

Reply:
good point